### PR TITLE
chown: added UResult

### PR DESCRIPTION
**Pull request by 353fc443**
Related to #2464

### CODE REVIEW

File: src/uu/chown/src/chown.rs [hunk 1]
```diff
@@ -131,10 +139,7 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
             Ok((None, Some(gid))) => IfFrom::Group(gid),
             Ok((Some(uid), Some(gid))) => IfFrom::UserGroup(uid, gid),
             Ok((None, None)) => IfFrom::All,
-            Err(e) => {
-                show_error!("{}", e);
-                return 1;
-            }
+            Err(e) => return Err(USimpleError::new(1, e)),
```
Comment (tertsdiepraam) [inline src/uu/chown/src/chown.rs:142]: Ideally, `parse_spec` would return `UResult`, so you don't need to create a custom error every time you call it. For example, this block would become:
```rust
match parse_spec(spec)? {
    (Some(uid), None) => IfFrom::User(uid),
    (None, Some(gid)) => IfFrom::Group(gid),
    (Some(uid), Some(gid)) => IfFrom::UserGroup(uid, gid),
    (None, None) => IfFrom::All,
}
```

File: src/uu/chown/src/chown.rs [hunk 2]
```diff
@@ -149,8 +154,10 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
                 dest_uid = Some(meta.uid());
             }
             Err(e) => {
-                show_error!("failed to get attributes of '{}': {}", file, e);
-                return 1;
+                return Err(USimpleError::new(
+                    1,
+                    format!("failed to get attributes of '{}': {}", file, e),
+                ));
```
Comment (tertsdiepraam) [inline src/uu/chown/src/chown.rs:160]: `fs::metadata` returns an `io::Result`, so I think this is a nice opportunity to use `UIoResult`:
```rust
let meta = fd::metadata(&file)
    .map_err_context(|| format!("failed to get attributes of '{}'"))?;
let dest_gid = Some(meta.gid());
let dest_uid = Some(meta.uid());
```

File: src/uu/chown/src/chown.rs [hunk 3]
```diff
@@ -160,8 +167,7 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
                 dest_gid = g;
             }
             Err(e) => {
-                show_error!("{}", e);
-                return 1;
+                return Err(USimpleError::new(1, e));
```
Comment (tertsdiepraam) [inline src/uu/chown/src/chown.rs:170]: When `parse_spec` returns `UResult`:
```rust
let (u, g) = parse_spec(owner)?;
dest_uid = u;
dest_gid = g;
```
(sadly, destructing assignments are unstable)

**Review by tertsdiepraam — CHANGES_REQUESTED** [review]
Thanks! I've left a few suggestions to clean it up some more and take full advantage of `UResult` :)

File: src/uu/chown/src/chown.rs [hunk 4]
```diff
@@ -7,13 +7,18 @@
 
 // spell-checker:ignore (ToDO) COMFOLLOW Chowner Passwd RFILE RFILE's derefer dgid duid
 
+// Clippy bug: https://github.com/rust-lang/rust-clippy/issues/7422
+#![allow(clippy::nonstandard_macro_braces)]
+
```
Comment (miDeb) [inline src/uu/chown/src/chown.rs:12]: Can you try removing this? The issue seems to have been fixed, maybe we don't even need this anymore :)

Comment (353fc443) [reply src/uu/chown/src/chown.rs]: Yeah removed

File: src/uu/chown/src/chown.rs [hunk 5]
```diff
@@ -330,12 +318,15 @@ macro_rules! unwrap {
...
+    fn exec(&self) -> UResult<()> {
         let mut ret = 0;
         for f in &self.files {
             ret |= self.traverse(f);
         }
-        ret
+        if ret != 0 {
+            return Err(UError::from(ret));
+        }
+        Ok(())
```
Comment (tertsdiepraam) [inline src/uu/chown/src/chown.rs:329]: This still feels awkward and made me realize that `traverse`, `obtain_meta` might also be nice to convert to `UResult`. You could even change `wrap_chown` if you want (although that is also used by `install`)

**Review by tertsdiepraam — APPROVED** [review]
The error handling still runs deeper than I initially thought :) If you want you could pick this up or you could leave it and I'll try to do it in a PR where I also change `wrap_chown` to use `UResult`.

**Comment by 353fc443** [comment]
@tertsdiepraam You can pick this up

**Pull request merged**